clients/src/test/java/org/apache/kafka/test/MockDeserializer.java

```diff
@@ -26,17 +26,16 @@
 
 public class MockDeserializer implements ClusterResourceListener, Deserializer<byte[]> {
     public static AtomicInteger initCount = new AtomicInteger(0);
-    public static AtomicInteger closeCount = new AtomicInteger(0);
     public static AtomicReference<ClusterResource> clusterMeta = new AtomicReference<>();
     public static ClusterResource noClusterId = new ClusterResource("no_cluster_id");
     public static AtomicReference<ClusterResource> clusterIdBeforeDeserialize = new AtomicReference<>(noClusterId);
 
     public boolean isKey;
     public Map<String, ?> configs;
+    public boolean isClosed = false;
```

Comment:
On another note, why are we switching to `boolean` from `integer`? I agree that there's no use of `closeCount` in the code base and that this is not public interface. But this changes the symmetry with `initCount` (which remains an integer) and it could still be used by a third-party package that uses these tests (by incorrectly depending on non-public interface). My point is, if we just do this for an assertion, we could use 

`closeCount.get() > 0` or `close.get() == 0` for the equivalent of `true` or `false` 

and be on the safe side (preserving symmetry too).

Reply:
I did not reuse the atomic count since the count is a static variable. Junit, by default, could run different tests on same JVM and hence the static variable may be changed by other tests.


For another, your point makes sense that the atomic integer is good enough. I will revert the Boolean and remove the static modifier to prevent the case I described above.

Reply:
I revert all changes of ```MockDeserializer``` as the concern I described is not existent yet.